### PR TITLE
Yizha/extend links header

### PR DESCRIPTION
**Pull request by yingzhao-1**

### CODE REVIEW

File: src/Sia.Gateway/Requests/Incidents/GetIncident.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using Sia.Domain;
 using Sia.Shared.Authentication;
 using Sia.Shared.Requests;
+using Sia.Shared.Protocol;
```
Comment (strand) [inline src/Sia.Gateway/Requests/Incidents/GetIncident.cs]: I don't see why this change is needed… 👇🏼 the revision below does not introduce a dependency on shared protocol.

File: src/Sia.Gateway/Initialization/Startup.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using Microsoft.AspNetCore.Builder;
+﻿ using Microsoft.AspNetCore.Builder;
```
Comment (strand) [inline src/Sia.Gateway/Initialization/Startup.cs]: Remove meaningless whitespace change.

File: test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            eventsController.CreateLinks("1", "2", null,null,"");
+
+            // Assert
+            urlHelperMock.Verify(foo => foo.Link(EventsController.GetSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(EventsController.PostSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(EventsController.GetMultipleRouteName, It.IsAny<object>()), Times.Exactly(1));
+            urlHelperMock.Verify(foo => foo.Link(IncidentsController.GetSingleRouteName, It.IsAny<object>()), Times.Exactly(1));
+
+            Assert.AreEqual(ids[0].ToString(), "{ id = 1 }");
+            //Assert.AreEqual(ids[0], new { id = "1" });
```
Comment (strand) [inline test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: delete comments.

File: test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            Assert.AreEqual(ids[1].ToString(), "{ }");
+            Assert.AreEqual(ids[2].ToString(), "{ }");
+            Assert.AreEqual(ids[3].ToString(), "{ id = 2 }");
```
Comment (strand) [inline test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: Looking closer at both CreateLinks methods I noticed that their argument signature is similar, but not the same, and they are. Design-wise I would prefer that the accepted arguments were the same, along with the format of the return values, though I think this sort of nitpickiness is overkill.

File: test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+        [TestMethod]
+        public void CreateLinksWorksCorrectly_WhenCalledInEventsMethods()
+        {
+            //Arrange
+            methods.Clear();
+            ids.Clear();
+            var eventsController = new EventsController(null, null, null, urlHelperMock.Object);
+            // Act
+            eventsController.CreateLinks("1", "2", null,null,"");
```
Comment (strand) [inline test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: Use named arguments in method invocation, as the order of arguments doesn't seem to indicate anything.

File: test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                    (s, o) =>
+                    {
+                        methods.Add(s);
+                        ids.Add(o);
+                    }
+                );
+
+        }
+
+        string GetId(object values, string property) => values.GetType().GetProperty(property)?.GetValue(values).ToString() ?? "";
```
Comment (macrobug) [inline test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: Since "id" is passed as a parameter, maybe GetProperty would be a better name?

File: src/Sia.Gateway/Requests/Incidents/GetIncident.cs (hunk 1 above)
Comment (pdimitratos) [reply src/Sia.Gateway/Requests/Incidents/GetIncident.cs]: https://stackoverflow.com/questions/283471/remove-unused-namespaces-across-a-whole-project-or-solution-at-once has an interesting solution to remove unnecessary usings across the whole solution; we may want to do this regularly or set up a travis rule or something that alerts us to unnecessary usings. They're relatively hard to detect manually and I know I don't often think about removing namespaces when I'm no longer using them (which may be its own issue)

File: test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                        ids.Add(o);
+                    }
+                );
+
+        }
+        
+        string GetProperty(object values, string property) => values.GetType().GetProperty(property)?.GetValue(values).ToString() ?? "";
+
+        [TestMethod]
+        public void CreateLinksWorksCorrectly_WhenCalledInEventsMethods()
```
Comment (pdimitratos) [inline test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: It's often better to be more descriptive in test names; we can assume that any test is verifying that something works correctly, and the test name should include a description of what behavior we consider 'working correctly' to be. Similarly, the 'when' statement should give information about the content of the context of the method call rather than where the method is being called (unless the 'when' materially changes the behavior of the method). This test might be better as 'CreateLinksGeneratesFourLinksWithCorrectIds_WhenPassedAnIncidentIdAndAnEventId()', but there are likely a number of other options at least as descriptive as that. Naming is a very hard problem, but effort put into it pays off over time.

Comment (yingzhao-1) [reply test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs]: CreateLinksGeneratesFourLinksWithCorrectIds_WhenPassedAnIncidentIdAndAnEventId() does make more sense.
